Userland/Services/RequestServer/ConnectionCache.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ConnectionCache.h"
+#include <LibCore/EventLoop.h>
+
+namespace RequestServer::ConnectionCache {
+
+HashMap<String, Vector<Connection<Core::TCPSocket>>> g_tcp_connection_cache {};
+HashMap<String, Vector<Connection<TLS::TLSv12>>> g_tls_connection_cache {};
```

Comment:
Would it make sense to key these by IP address instead of hostname?

Reply:
Nope, lots of servers like to use SNI for virtual hosts, so there's no real IP<=>hostname mapping in TLS.